Plugins/GitUIPluginInterfaces/ViewModels/GitRepoViewModel.cs

```diff
@@ -0,0 +1,32 @@
+﻿using System.ComponentModel;
+using System.ComponentModel.Design;
+using CommunityToolkit.Mvvm.ComponentModel;
+using CommunityToolkit.Mvvm.Messaging;
+using GitUIPluginInterfaces;
+
+namespace Plugins.GitUIPluginInterfaces.ViewModels;
+
+public interface IGitRepoViewModel : INotifyPropertyChanged
+{
+    IEnumerable<GitRevision> Revisions { get; }
+}
+
+internal partial class GitRepoViewModel : ObservableObject, IGitRepoViewModel, IRecipient<RepoModifiedMessage>
+{
+    internal static void Register(IServiceContainer serviceContainer)
+    {
+        IGitRepoViewModel model = new GitRepoViewModel();
+        serviceContainer.AddService(model);
```

Comment:
❗ This feels like we're not controlling the registration surface. The registrations should only be performed at the bootstrap of the app and constrained to `ServiceContainerRegistry` classes.

Reply:
changed